plugins/modules/notification/discord.py

```diff
@@ -0,0 +1,172 @@
+  community.general.discord:
+    webhook_id: "00000"
+    webhook_token: "XXXYYY"
+    content: "This is a message from ansible"
+
+- name: Send a message to the discord channel with specific username and avatar
+  community.general.discord:
+    webhook_id: "00000"
+    webhook_token: "XXXYYY"
+    content: "This is a message from ansible"
+    username: Ansible
+    avatar_url: "https://docs.ansible.com/ansible/latest/_static/images/logo_invert.png"
+
+- name: Send a embedded message to the discord channel
+  community.general.discord:
+    webhook_id: "00000"
+    webhook_token: "XXXYYY"
+      embeds:
+        - title: "Embedded message"
+          description: "This is an embedded message"
+          footer:
+            text: "Author: Ansible"
+          image:
+            url: "https://docs.ansible.com/ansible/latest/_static/images/logo_invert.png"
+"""
+
+RETURN = """
+http_code:
+  description:
+    - Response Code returned by discord API.
+  returned: always
+  type: int
+  sample: 204
+"""
+
+
+def discord_text_msg(module):
+
+    webhook_id = module.params['webhook_id']
+    webhook_token = module.params['webhook_token']
+    content = module.params['content']
+    user = module.params['username']
+    avatar_url = module.params['avatar_url']
+    tts = module.params['tts']
+    embeds = module.params['embeds']
+
+    headers = {
+        'content-type': 'application/json'
+    }
+
+    url = "https://discord.com/api/webhooks/%s/%s" % (
+        webhook_id, webhook_token)
+
+    payload = {
+        'content': content,
+        'username': user,
+        'avatar_url': avatar_url,
+        'tts': tts,
+        'embeds': embeds,
+    }
+
+    payload = module.jsonify(payload)
+
+    response, info = fetch_url(module, url, data=payload, headers=headers)
+    return response, info
+
+
+def main():
+    module = AnsibleModule(
+        argument_spec=dict(
+            webhook_id=dict(type='str', required=True),
+            webhook_token=dict(type='str', required=True, no_log=True),
+            content=dict(type='str', required=False),
+            username=dict(type='str', required=False),
+            avatar_url=dict(type='str', required=False),
+            tts=dict(type='bool', required=False, default=False),
+            embeds=dict(type='list', required=False),
+        ),
+        required_one_of=[['content', 'embeds']],
+        supports_check_mode=True
+    )
+
+    result = dict(
+        changed=False,
+        stdout='',
+        stderr='',
+        http_code='',
+        rc=0
+    )
+
+    if module.check_mode:
+        module.exit_json(changed=False)
+    else:
+        response, info = discord_text_msg(module)
+
+    if info['status'] != 204:
+        result['http_code'] = info['status']
+        result['stderr'] = info['msg']
+        result['failed'] = True
+        result['rc'] = 1
+        module.fail_json(msg="Failed to send message",
+                         info=info, response=response, result=result)
```

Comment:
`rc`, `stdout` and `stderr` are usually associated with file descriptors in processes. As this module does not spawn a child process - it performs its job by placing a HTTP request instead - this terminology sounds a bit off. Maybe it might make more sense to pass `info['msg']` in the `fail_json()` call?
My suggestion would be:
```suggestion
        result['http_code'] = info['status']
        result['msg'] = info['msg']
        result['response'] = response,
        module.fail_json(**result)
```
Naturally you might need to adjust some of the tests for this output structure.

Reply:
Yes, you're right.
This thoughts comes from my other module, which uses cmd to spawn a child process. Removed/changed the output.